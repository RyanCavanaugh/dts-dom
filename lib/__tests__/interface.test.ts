import * as dom from "../index";

describe("an emitted interface", () => {
  describe("empty", () => {
    it("matches the snapshot", () => {
      let iface = dom.create.interface("Interface");

      expect(dom.emit(iface)).toMatchSnapshot();
    });
  });
  describe("with type parameters", () => {
    it("matches the snapshot", () => {
      let clazz = dom.create.class("BaseFoo");
      let iface = dom.create.interface("Interface");
      iface.typeParameters.push(dom.create.typeParameter("T1"));
      iface.typeParameters.push(
        dom.create.typeParameter("T2", dom.type.object)
      );
      iface.typeParameters.push(dom.create.typeParameter("T3", clazz));

      expect(dom.emit(iface)).toMatchSnapshot();
    });
  });
});
