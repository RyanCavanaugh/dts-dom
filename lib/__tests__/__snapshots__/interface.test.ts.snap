// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an emitted interface empty matches the snapshot 1`] = `
declare interface Interface {
}


`;

exports[`an emitted interface with type parameters matches the snapshot 1`] = `
declare interface Interface<T1, T2 extends object, T3 extends BaseFoo> {
}


`;
